Mathlib/RingTheory/TwoSidedIdeal.lean

```diff
@@ -0,0 +1,125 @@
+/-
+Copyright (c) 2024 Jujian Zhang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jujian Zhang
+-/
+
+import Mathlib.RingTheory.Congruence
+import Mathlib.Algebra.Module.LinearMap.Basic
+import Mathlib.Tactic.Abel
+
+/-!
+# Two Sided Ideals
+
+In this file, for any `Ring R`, we reinterpret `I : RingCon R` as a two sided ideal of a ring.
+
+-/
+
+section ideal
+
+namespace RingCon
+
+variable {R : Type*} [Ring R]
+
+instance : SetLike (RingCon R) R where
+  coe t := {r | t r 0}
```

Comment:
Can you add a (non-`simp`) lemma `mem_iff` that expands membership to `t r 0`?

Reply:
done in here [cfe1bbd](https://github.com/leanprover-community/mathlib4/pull/13401/commits/cfe1bbdc7aca2906bf8022bdca025714576617ca)